emote/memory/memory.py

```diff
@@ -209,13 +212,72 @@ def add(
 
         return self._inner.add(observations, responses)
 
+    def report(
+        self,
+        metrics: Dict[str, float],
+        metrics_lists: Dict[str, List[float]],
+    ):
+        if "agent_metrics" in metrics:
+            agent_metrics = metrics.pop("agent_metrics")
```

Comment:
Agent metrics is the one part we already handle - the agent_metrics list is set up as 
`agent_metrics = [x.metadata.info for x in observations.values()]` , which is the loop on line 196. So you can replace the logging in that loop with `self.report(obs.metadata.info, obs.metadata.info_lists)`, and do the histogram-branch below.